backend/src/api/mempool-blocks.ts

```diff
@@ -321,6 +327,96 @@ class MempoolBlocks {
     }
   }
 
+  private resetRustGbt(): void {
+    this.rustInitialized = false;
+    this.rustGbtGenerator = new GbtGenerator();
+  }
+
+  private async $rustMakeBlockTemplates(newMempool: { [txid: string]: MempoolTransactionExtended }, saveResults: boolean = false): Promise<MempoolBlockWithTransactions[]> {
+    const start = Date.now();
+
+    // reset mempool short ids
+    if (saveResults) {
+      this.resetUids();
+    }
+    // set missing short ids
+    for (const tx of Object.values(newMempool)) {
+      this.setUid(tx, !saveResults);
+    }
+
+    // serialize relevant mempool data into an ArrayBuffer
+    // to reduce the overhead of passing this data to the rust thread
+    const mempoolBuffer = this.mempoolToArrayBuffer(Object.values(newMempool), newMempool);
+
+    // run the block construction algorithm in a separate thread, and wait for a result
+    const rustGbt = saveResults ? this.rustGbtGenerator : new GbtGenerator();
+    try {
+      const { blocks, rates, clusters } = this.convertNapiResultTxids(
+        await rustGbt.make(new Uint8Array(mempoolBuffer)),
+      );
+      if (saveResults) {
+        this.rustInitialized = true;
+      }
+      const processed = this.processBlockTemplates(newMempool, blocks, rates, clusters, saveResults);
+      logger.debug(`RUST makeBlockTemplates completed in ${(Date.now() - start)/1000} seconds`);
+      return processed;
+    } catch (e) {
+      logger.err('RUST makeBlockTemplates failed. ' + (e instanceof Error ? e.message : e));
+      if (saveResults) {
+        this.resetRustGbt();
+      }
+    }
+    return this.mempoolBlocks;
+  }
+
+  public async $oneOffRustBlockTemplates(newMempool: { [txid: string]: MempoolTransactionExtended }): Promise<MempoolBlockWithTransactions[]> {
+    return this.$rustMakeBlockTemplates(newMempool, false);
+  }
+
+  public async $rustUpdateBlockTemplates(newMempool: { [txid: string]: MempoolTransactionExtended }, added: MempoolTransactionExtended[], removed: MempoolTransactionExtended[]): Promise<void> {
+    // sanity check to avoid approaching uint32 uid overflow
+    if (this.nextUid > 4_000_000_000) {
+      this.resetRustGbt();
+    }
```

Comment:
Why not `4_294_967_295`? (Exactly u32::MAX)

Reply:
just to leave a margin of error for the added transactions (obviously there won't ever be 300m new transactions, but if the threshold isn't exactly u32::MAX it might as well be a round number)

I suppose

```
if (this.nextUid + added.length > u32::MAX) {
  this.resetRustGbt();
}
```
would be convey the intent more clearly